### PR TITLE
Implemented the ThreadedContextManager to support synchronous resources

### LINKED ISSUE #13: Implement the ThreadedContextManager for synchronous resources
Since synchronous resources are by definition synchronous, we need to acquire & release them using a thread pool.
Therefore, we must wrap them in a `ThreadedContextManager` which schedules `__enter__` and `__exit__` in the thread pool.

The `ThreadedContextManager` follows the classic [Adapter design pattern](https://refactoring.guru/design-patterns/adapter).

By default, we use a global [`CapacityLimiter`](https://anyio.readthedocs.io/en/stable/synchronization.html#capacity-limiters) passed into the [`anyio.run_sync_in_worker_thread()`](https://anyio.readthedocs.io/en/stable/api.html#anyio.run_sync_in_worker_thread).
The capacity limiter is set to the current number of CPUs.

You may optionally pass another `CapacityLimiter` instance.

### PR DESCRIPTION
**Pull request by thedrow**
Fixes #13.

### CODE REVIEW

File: jumpstarter/resources.py [hunk 1]
```diff
@@ -89,6 +91,24 @@ def __get__(self, instance, owner):
             return instance._resources[self._name]
 
 
+class ThreadedContextManager(ObjectProxy):
+    def __init__(
+        self, context_manager: typing.ContextManager, capacity_limiter: CapacityLimiter
+    ):
+        super(ThreadedContextManager, self).__init__(context_manager)
```
Comment (ahopkins) [inline jumpstarter/resources.py:98]: Since this is targeting newer versions of Python, why the verbosity?

**Review by ahopkins — APPROVED** [review]

File: poetry.lock [hunk 2]
```diff
@@ -383,6 +383,14 @@ category = "main"
 optional = false
```
Comment (auvipy) [inline poetry.lock:383]: great job for introducing poetry

**Review by auvipy — APPROVED** [review]
except for the explicit super(ThreadedContextManager, self) other codes look good to me

File: jumpstarter/resources.py (hunk 1 above)
Comment (thedrow) [reply jumpstarter/resources.py:98]: Oh right! That's how PyCharm autocompletes it, unfortunately.

**Pull request merged**